servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/PublisherFlatMapSingleTest.java

```diff
@@ -639,4 +644,23 @@ void testRequestAndEmitConcurrency() throws Exception {
         assertThat("Unexpected items emitted.", received, hasSize(totalToRequest));
         assertThat(received, containsInAnyOrder(range(0, totalToRequest).boxed().toArray()));
     }
+
+    @Test
+    void testConcurrentEmitions() throws ExecutionException, InterruptedException {
+        final int maxSingles = 1_000;
+        final Executor executor = io.servicetalk.concurrent.api.Executors.newFixedSizeExecutor(10);
```

Comment:
1. Executor have to be closed/shutdown after the test. Do it manually or consider using `ExecutorExtension` to automate the cleaning.
2. Consider using static import.